src/component/utils/get-name-for-type/index.ts

```diff
@@ -1,10 +1,15 @@
 import { startCase } from 'lodash/fp';
 
-type Type = {
-  name: string;
-  title?: string;
-};
+import { Field, Type } from '../../types';
+
+type FieldOrType = Field | Type;
 
-const getNameForType = (type: Type): string => type.title || startCase(type.name);
+const getNameForType = (type: FieldOrType): string => {
+  if ('title' in type) {
+    return type.title ?? '';
+  }
```

Comment:
hows this work?
do we ever want this `''` case?

Reply:
It'll never hit it because title is always defined, but without the `?? ''` case it throws compiler errors.